nThElemLinkedList.py

```diff
@@ -1,48 +1,68 @@
 """
 Implement an algorithm to find the kth to last element of a singly linked list.
+"""
+class Node(object):
+    # Function to initialize the node object
+    def __init__(self, data=None, next_node=None):
+        self.data = data  # Assign data
+        self.next_node = next_node  # Initialize 
+                          # next as null
+    def get_data(self):
+        return self.data
 
-The first approach is to loop through the list so we could get the lenght of the linked list ,N.
-Next we can see the element K th to the last element using formula : ( length(list)- k + 1 )
+    def get_next(self):
+        return self.next_node
 
-Complexity :
-Time complexicity: O(k)
-Space complexicity: O(1)
+    def set_next(self, new_next):
```

Comment:
Likewise as above. Though there are decorators that allow hiding getters/setters behind method calls if there is special logic to be executed at get/set time:

https://docs.python.org/3.6/library/functions.html#property